src/python/pants/backend/python/tasks2/python_execution_task_base.py

```diff
@@ -130,6 +116,9 @@ def create_pex(self, pex_info=None):
 
         extra_pex_paths = [pex.path() for pex in pexes if pex]
 
+        if extra_pex_paths:
+          pex_info.merge_pex_path(':'.join(extra_pex_paths))
+
         with safe_concurrent_creation(path) as safe_path:
           builder = PEXBuilder(safe_path, interpreter, pex_info=pex_info)
           builder.freeze()
```

Comment:
also seems like you no longer need the `extra_pex_paths_file_path` handling down below (lines 126-133) both inside and outside the invalidated block, since `WrappedPex` no longer does anything with that and it should all be written directly into the pex/chroot's `PEX-INFO`.